File: cpp/leetcode/167.cpp
class Solution {
  public:
    vector<int> twoSum(vector<int> &nums, int target) {
        if (nums.size() < 2) {
            throw invalid_argument("Invalid nums");
        }
        vector<int> res;
        int l = 0, r = nums.size() - 1;
        while (l < r) {
            if (nums[l] + nums[r] == target) {
                res.push_back(l + 1);
                res.push_back(r + 1);
                return res;
            } else if (nums[l] + nums[r] > target) {
                r--;
            } else {
                l++;
            }
        }
        throw invalid_argument("Can't find solution");
    }
};

File: python/leetcode/167.py
class Solution(object):
    def twoSum(self, nums, target):
        if len(nums) < 2:
            raise ValueError("Invalid nums array")
        l, r = 0, len(nums) - 1
        while l < r:
            if nums[l] + nums[r] == target:
                return [l + 1, r + 1]
            elif nums[l] + nums[r] > target:
                r -= 1
            else:
                l += 1